### PR TITLE
feat: Add NearestElement stack algorithms and fix Javadoc build warnings

### LINKED ISSUE #7322: Permutations in java
### What would you like to Propose?

I would like to add few programs in java that uses recursion. So first I want to start with permutations.

### Issue details

Currently under recursions,there are few programs,I would like to add more 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sabarirajan83**
### Description
This PR introduces `NearestElement.java`, which provides stack-based algorithms to efficiently find the nearest greater or smaller elements in an array (for both left and right directions). 
* Implements a Monotonic Stack approach to achieve O(n) time complexity.
* Includes explicit safety checks (using short-circuit evaluation and ternary operators) to prevent `EmptyStackException`.
* Includes comprehensive JUnit 5 tests in `NearestElementTest.java`.

Additionally, this PR fixes pre-existing `documentation comment is not attached to any declaration` errors in:
- `AnyBaseToAnyBase.java`
- `InterpolationSearch.java`
- `LinearSearch.java`

These documentation changes ensure strict build compatibility with modern JDK versions (JDK 21+).

Fixes #7322

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations. 
- [x] All new algorithms include a corresponding test class that validates their functionality.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/datastructures/stacks/NearestElement.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+     * @throws IllegalArgumentException if the input array is null
+     */
+    public static int[] nearestGreaterToRight(int[] arr) {
+        if (arr == null) {
+            throw new IllegalArgumentException("Input array cannot be null");
+        }
+        int n = arr.length;
+        int[] result = new int[n];
+        Stack<Integer> stack = new Stack<>();
+
```
Comment (prashantpiyush1111) [inline src/main/java/com/thealgorithms/datastructures/stacks/NearestElement.java:33]: Consider using ArrayDeque instead of Stack class, 
it is faster and recommended in Java docs.

File: src/test/java/com/thealgorithms/datastructures/stacks/NearestElementTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        // Smaller to Right: MAX -> 0, 0 -> MIN, MIN -> none
+        assertArrayEquals(new int[]{0, Integer.MIN_VALUE, -1}, NearestElement.nearestSmallerToRight(input));
+        // Greater to Left: MAX -> none, 0 -> MAX, MIN -> 0
+        assertArrayEquals(new int[]{-1, Integer.MAX_VALUE, 0}, NearestElement.nearestGreaterToLeft(input));
+    }
+
+    // --- Exception Handling ---
+
+    @Test
+    void testNullInput() {
```
Comment (prashantpiyush1111) [inline src/test/java/com/thealgorithms/datastructures/stacks/NearestElementTest.java:104]: Great test coverage for null input edge case!

**Review by prashantpiyush1111 — COMMENTED** [review]
Good implementation overall! Reviewed the NearestElement 
stack algorithm. Suggest using ArrayDeque instead of 
Stack class for better performance.

**Comment by DenizAltunkapan** [comment]
too many files changed and no contribution in last 2 weeks. Also conflicts are existing.
Please look into our [issues](https://github.com/TheAlgorithms/Java/issues) if you need other contribution ideas
